### PR TITLE
add terraformBoot healthCheck to xpanse healthCheck background

### LINKED ISSUE #774: add terraform-boot health check to xpanse healthcheck background
- [ ] use the /health method of the terraform-boot to check if terraform boot is API is reachable.
- [ ] Add terraform-boot backend to the current list of backends that checked in Xpanse's healthcheck API

### PR DESCRIPTION
**Pull request by jinyangyang222**
fixes #774

### CODE REVIEW

File: modules/api/src/main/java/org/eclipse/xpanse/api/AdminServicesApi.java [hunk 1]
```diff
@@ -46,6 +49,8 @@
 @Secured({ROLE_ADMIN})
 public class AdminServicesApi {
 
+    private final TerraformApi terraformApi = new TerraformApi();
```
Comment (swaroopar) [inline modules/api/src/main/java/org/eclipse/xpanse/api/AdminServicesApi.java:52]: @jinyangyang222 I see there is some issue with the generated classes. The generated classes must be spring beans and simply be `autowired`. 
<generatorName>java</generatorName> -> must be `spring` for the classes to be generated as spring beans.

Can you please check this?

File: modules/api/src/main/java/org/eclipse/xpanse/api/AdminServicesApi.java [hunk 2]
```diff
@@ -89,10 +94,39 @@ private List<BackendSystemStatus> checkHealthOfBackendSystem() {
...
+    }
+
+    private BackendSystemStatus getTerraformBootStatus() {
+        BackendSystemStatus terraformBootStatus = new BackendSystemStatus();
+        terraformBootStatus.setBackendSystemType(BackendSystemType.TERRAFORM_BOOT);
+        terraformBootStatus.setName(BackendSystemType.TERRAFORM_BOOT.toValue());
+        if (isTerraformBootApiAccessible()) {
+            terraformBootStatus.setHealthStatus(HealthStatus.OK);
+            ApiClient apiClient = new ApiClient();
+            terraformBootStatus.setEndpoint(apiClient.getBasePath());
```
Comment (swaroopar) [inline modules/api/src/main/java/org/eclipse/xpanse/api/AdminServicesApi.java:123]: we must also customize the base path config. 

The basepath must be read from spring-properties and then added to the ApiClient. 

The generated class has the default url and this must not be directly used.

**Review by swaroopar — CHANGES_REQUESTED** [review]
comments added.

File: modules/api/src/main/java/org/eclipse/xpanse/api/AdminServicesApi.java (hunk 1 above)
Comment (jinyangyang222) [reply modules/api/src/main/java/org/eclipse/xpanse/api/AdminServicesApi.java:52]: ok,I'll check this.

File: modules/api/src/main/java/org/eclipse/xpanse/api/AdminServicesApi.java [hunk 3]
```diff
@@ -46,12 +48,27 @@
...
+    private final TerraformApi terraformApi;
+
+    private final ApiClient apiClient;
+
+    /**
+     * The constructor.
+     */
+    public AdminServicesApi(IdentityProviderManager identityProviderManager,
+                            TerraformApi terraformApi, ApiClient apiClient,
+                            @Value("${apiClient.basePath:}") String baseUrl) {
```
Comment (swaroopar) [inline modules/api/src/main/java/org/eclipse/xpanse/api/AdminServicesApi.java:65]: @jinyangyang222 we must not update the basepath here. Instead we need to update the bean directly at startup and simply use it everywhere, 
example - https://github.com/Zomzog/openapi-spring-client-token/blob/master/front/src/main/java/bzh/zomzog/demo/openapispringclienttoken/front/config/BackendClientConfig.java

File: runtime/src/main/resources/application-terraformboot-local.properties [hunk 4]
```diff
@@ -0,0 +1,6 @@
+#
+# SPDX-License-Identifier: Apache-2.0
+# SPDX-FileCopyrightText: Huawei Inc.
+#
+#
+apiClient.basePath=http://localhost:9090
```
Comment (swaroopar) [inline runtime/src/main/resources/application-terraformboot-local.properties:6]: property name - terraformboot.endpoint

File: runtime/src/main/resources/application-terraformboot-local.properties [hunk 5]
```diff
@@ -0,0 +1,6 @@
+#
```
Comment (swaroopar) [inline runtime/src/main/resources/application-terraformboot-local.properties:1]: we can just have one profile - application-terraformboot.properties and enable it by default. 
Pls also check once with WangLi.. i think he also has added a new profile in this PR.

**Review by swaroopar — CHANGES_REQUESTED** [review]
comments added

File: modules/api/src/main/java/org/eclipse/xpanse/api/AdminServicesApi.java (hunk 3 above)
Comment (jinyangyang222) [reply modules/api/src/main/java/org/eclipse/xpanse/api/AdminServicesApi.java]: ok,fixed

**Review by swaroopar — APPROVED** [review]
LGTM

**Pull request merged**